File: SkillSurvey.Entities/JobPost.js
class JobPost {
    constructor(parameters) {
        this.Id = parameters.Id;
        this.SiteId = parameters.SiteId;
        this.JobSiteNumber = parameters.JobSiteNumber
        this.Title = parameters.Title;
        this.Body = parameters.Body;
        this.DatePosted = parameters.DatePosted;
        this.City = parameters.City;
        this.Country = parameters.Country;
        this.Suburb = parameters.Suburb;
    }
}

exports.New = (parameters) => new JobPost(parameters);

File: pkg/entities/jobpost.go
package entities

import (
	"time"
)

type JobPost struct {
	SiteId        int
	JobSiteNumber string
	Title         string
	Body          string
	PostedDate    time.Time
	City          string
	Country       string
	Suburb        string
}

type InboundJobPost struct {
	JobPost
	SiteName string
}
